pkg/api/addons.go

```diff
@@ -680,6 +666,16 @@ func (cs *ContainerService) setAddonsConfig(isUpgrade bool) {
 		},
 	}
 
+	if !cs.Properties.IsAzureStackCloud() {
+		defaultCorednsAddonsConfig.Config["cores-per-replica"] = "256"
```

Comment:
These defaults derived from the official documentation here:

https://kubernetes.io/docs/tasks/administer-cluster/dns-horizontal-autoscaling/